packages/contracts/contracts/Semaphore.sol

```diff
@@ -150,15 +104,15 @@ contract Semaphore is ISemaphore, SemaphoreGroups {
     function verifyProof(
         uint256 groupId,
         uint256 merkleTreeRoot,
-        uint256 signal,
-        uint256 nullifierHash,
-        uint256 externalNullifier,
+        uint256 nullifier,
+        uint256 message,
+        uint256 scope,
         uint256[8] calldata proof
-    ) external override {
+    ) external override onlyExistingGroup(groupId) {
         uint256 merkleTreeDepth = getMerkleTreeDepth(groupId);
```

Comment:
This merkle tree depth is 0 when there's only 1 group member. The merkle root is simply the identity commitment at that time.

This just needs to be changed to `getMerkleTreeSize` instead and it will  work as intended.

See the [definition in the zk-kit imt.sol repo](https://github.com/privacy-scaling-explorations/zk-kit/blob/main/packages/imt.sol/contracts/internal/InternalLeanIMT.sol#L8)

Reply:
Thank you @numtel. Please, see #535 to reflect your requested changes